### PR TITLE
fix(issue-views): Fix query being overwritten on refresh of default tab

### PR DESCRIPTION
**Pull request by MichaelSun48**
Fixes an issue where attempting to navigate to a predefined query with only a default tab would redirect you to the first tab and overwrite the original query.

### CODE REVIEW

**Comment by malwilley** [comment]
@MichaelSun48 looks like there's a failing test

File: static/app/views/issueList/customViewsHeader.spec.tsx [hunk 1]
```diff
@@ -265,6 +265,57 @@ describe('CustomViewsHeader', () => {
         })
       );
     });
+
+    it('initially selects a temporary tab if the viewId is a default viewId and the query is different', () => {
```
Comment (malwilley) [inline static/app/views/issueList/customViewsHeader.spec.tsx:269]: Is this the behavior that we want? For saved tabs it keeps you on the same tab and displays unsaved changes so I'd expect it to work the same for an unsaved tab, right?

Comment (MichaelSun48) [reply static/app/views/issueList/customViewsHeader.spec.tsx:269]: Hmm yeah you're definitely right about this - redirecting to a temporary tab does conflict with what we would expect from any other tab. Will make that change now.

File: static/app/views/issueList/customViewsHeader.tsx [hunk 2]
```diff
@@ -162,15 +162,6 @@ function CustomViewsIssueListHeaderTabsContent({
...
   const getInitialTabKey = () => {
-    if (draggableTabs[0].key.startsWith('default')) {
-      if (query) {
-        return TEMPORARY_TAB_KEY;
-      }
-      return draggableTabs[0].key;
-    }
-    if (!query && !sort && !viewId) {
-      return draggableTabs[0].key;
-    }
```
Comment (MichaelSun48) [inline static/app/views/issueList/customViewsHeader.tsx:173]: Hopefully this simplifies the initial tab key logic a bit. Now the logic is:

1. If the query params contain a `viewId` **and** that it matches the id of one of the user's tabs, then that tab is the initial tab no matter what.
2. If the query params contain a `query`, then the initial tab is a temporary tab
3. For all other scenarios, the first tab is the initial tab.

Comment (malwilley) [reply static/app/views/issueList/customViewsHeader.tsx:173]: Nice! Removing lines to fix bugs, nothing better than that

**Review by malwilley — APPROVED** [review]

**Review by edwardgou-sentry — APPROVED** [review]

**Pull request merged**